### PR TITLE
32 sub workflow for gene annotation

### LINKED ISSUE #32: Sub-workflow for gene annotation
### Description of feature

Create a sub-workflow of gene annotation using the following modules from nf-core

- [x] [Bakta](https://github.com/nf-core/modules/tree/master/modules/nf-core/bakta)
- [x] [Prokka](https://github.com/nf-core/modules/tree/master/modules/nf-core/prokka)

### PR DESCRIPTION
**Pull request by anwarMZ**
A sub-workflow that contains `prokka` & `bakta` modules from nf-core. The parameters exposed are added to the `nextflow.config`. The `modules.json` is updated for the new modules. 
Both the annotation tools output gff files when required to connect to downstream analyses. All other output files are available through the `outdir`
The sub-workflow allows running prokka or bakta or both on a provided genome. There is an option to skip downloading bakta database is already provided using the config file. 

This pull request closes #32

### CODE REVIEW

File: subworkflows/local/gene_annotation.nf [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        prokka_gff = Channel.empty()
+        bakta_gff = Channel.empty()
+        ch_versions = Channel.empty()
+
+        // Checking if the prodigal training file is provided
+        if (!params.prodigal_training_file){
+            prodigal = []
+        }
+        else{
+            prodigals = file(params.prodigal_training_file, checkIfExists: true)
```
Comment (jimmyliu1326) [inline subworkflows/local/gene_annotation.nf:25]: `prodigal_training_file` parameter should be initialized in `nextflow.config`

File: subworkflows/local/gene_annotation.nf [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        // Checking if the annotation protein file is provided
+        if (!params.annotation_protein_file){
+            proteins = []
+        }
+        else{
+            proteins = file(params.annotation_protein_file, checkIfExists: true)
```
Comment (jimmyliu1326) [inline subworkflows/local/gene_annotation.nf:33]: `annotation_protein_file` parameter should be initialized in `nextflow.config`

File: subworkflows/local/gene_annotation.nf [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            
+            BAKTA_BAKTA(ch_genome, bakta_db, proteins, prodigal)
+            bakta_gff = BAKTA_BAKTA.out.gff
+            ch_versions = ch_versions.mix(BAKTA_BAKTA.out.versions)
+        }
+        
+    // Defining the output channels
+    emit:
+        bakta_gff
+        prokka_gff
```
Comment (jimmyliu1326) [inline subworkflows/local/gene_annotation.nf:65]: The subworkflow should also emit gene and protein sequences in FASTA format to enable downstream functional annotations e.g. by `eggnog-mapper`

**Review by jimmyliu1326 — CHANGES_REQUESTED** [review]

File: subworkflows/local/gene_annotation.nf [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            PROKKA(ch_genome, proteins, prodigal)
+            prokka_gff = PROKKA.out.gff
+            ch_versions   = ch_versions.mix(PROKKA.out.versions)
+        }
+
+        // Running BAKTA if skip_bakta is not set to true
+
+        if(!params.skip_bakta){
+            // Checking if BAKTA download is skipped
+            if (!params.skip_bakta_download){
```
Comment (jimmyliu1326) [inline subworkflows/local/gene_annotation.nf:47]: I would update the bakta database download condition to `!params.bakta_db` such that the database download is skipped when users supply a bakta database path.

**Review by jimmyliu1326 — CHANGES_REQUESTED** [review]

File: subworkflows/local/gene_annotation.nf (hunk 1 above)
Comment (anwarMZ) [reply subworkflows/local/gene_annotation.nf:25]: resolved by [c6d85a7](https://github.com/cidgoh/nf-seqqc/pull/44/commits/c6d85a7a7fac79b49086c954d56d6b8dd1251216)

File: subworkflows/local/gene_annotation.nf (hunk 2 above)
Comment (anwarMZ) [reply subworkflows/local/gene_annotation.nf:33]: resolved by [c6d85a7](https://github.com/cidgoh/nf-seqqc/pull/44/commits/c6d85a7a7fac79b49086c954d56d6b8dd1251216)

File: subworkflows/local/gene_annotation.nf (hunk 3 above)
Comment (anwarMZ) [reply subworkflows/local/gene_annotation.nf:71]: resolved with https://github.com/cidgoh/nf-seqqc/pull/44/commits/c00bb1cbc90547538197b3edb5fbf750d8bb3114

File: subworkflows/local/gene_annotation.nf (hunk 4 above)
Comment (anwarMZ) [reply subworkflows/local/gene_annotation.nf]: resolved with https://github.com/cidgoh/nf-seqqc/pull/44/commits/def6ab313e15397132d59f2f007cc31f816a5e95

File: subworkflows/local/gene_annotation.nf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+// Defining the workflow
+workflow GENE_ANNOTATION{
+    // Defining the input channel
+    take:
+        ch_genome
+
+    // Defining the main process
+    main:
+        // Initializing empty channels for the output files
```
Comment (jimmyliu1326) [inline subworkflows/local/gene_annotation.nf:15]: need to initialize channels for `prokka_fna`, `prokka_faa`, `bakta_fna`, `bakta_faa`

**Review by jimmyliu1326 — CHANGES_REQUESTED** [review]

File: subworkflows/local/gene_annotation.nf (hunk 5 above)
Comment (anwarMZ) [reply subworkflows/local/gene_annotation.nf:15]: resolved with [39ac2ca](https://github.com/cidgoh/nf-seqqc/pull/44/commits/39ac2ca1315a9cf9005206e59ddc84265cc88b21)

**Comment by anwarMZ** [comment]
Since there were no major changes asked except the missing empty channels (resolved with [39ac2ca](https://github.com/cidgoh/nf-seqqc/pull/44/commits/39ac2ca1315a9cf9005206e59ddc84265cc88b21)) I am going to merge this to save time.

**Pull request merged**